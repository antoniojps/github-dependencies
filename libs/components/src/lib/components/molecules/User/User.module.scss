.user {
  display: grid;
  grid-template-columns: auto 1fr;
  align-items: center;

  &Name {
    font-size: var(--size-xs);
    font-weight: var(--weight-bold);
    margin-right: var(--space-3x);
  }

  &Large {
    font-size: var(--size-xl4);
    word-break: break-all;
  }

  &Reverse {
    direction: rtl;
    text-align: left;

    & span {
      margin-right: 0;
      margin-left: var(--space-3x);
    }
  }
}
