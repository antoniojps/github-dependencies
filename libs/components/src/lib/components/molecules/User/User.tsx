import React, { ReactElement } from 'react';
import { Avatar } from '@geist-ui/react';
import styles from './User.module.scss';
import classNames from 'classnames';
import { capitalize } from 'lodash';

export type Props = {
  name: string;
  image?: string | null;
  hasName?: boolean;
  isReverse?: boolean;
  size?: 'large';
  className?: string;
};

export const User = ({
  name,
  image,
  hasName = true,
  isReverse = false,
  size,
  className = '',
}: Props): ReactElement => {
  const [firstLetter] = name.split('');

  return (
    <span
      className={classNames(styles.user, {
        [styles.userReverse]: isReverse,
        [className]: Boolean(className),
      })}
    >
      {hasName && (
        <span
          className={classNames(styles.userName, {
            ...(size ? { [styles[`user${capitalize(size)}`]]: true } : {}),
          })}
        >
          {name}
        </span>
      )}
      <Avatar src={image || undefined} text={firstLetter} />
    </span>
  );
};

export default User;
