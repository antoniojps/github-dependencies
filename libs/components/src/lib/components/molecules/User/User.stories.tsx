import React from 'react';
import User from './User';

export default {
  title: 'molecules/User',
};

export const Basic = () => (
  <User image={'https://react.geist-ui.dev/images/avatar.png'} name={'Joe'} size={'large'} />
);

export const WithoutImage = () => <User name="Antonio Santos" />;
