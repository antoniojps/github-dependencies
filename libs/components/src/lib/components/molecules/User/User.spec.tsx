import React from 'react';
import { render } from '@testing-library/react';
import { User } from './User';

describe('User', () => {
  it('renders', () => {
    const { asFragment } = render(<User name="Antonio" />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders with image', () => {
    const { asFragment } = render(
      <User name="Antonio" image="https://react.geist-ui.dev/images/avatar.png" />
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders with name reverse', () => {
    const { asFragment } = render(<User name="Antonio" isReverse />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders with size large', () => {
    const { asFragment } = render(<User name="Antonio" size="large" />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders with image, reverse and large', () => {
    const { asFragment } = render(
      <User name="Antonio" image="https://react.geist-ui.dev/images/avatar.png" size="large" />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
