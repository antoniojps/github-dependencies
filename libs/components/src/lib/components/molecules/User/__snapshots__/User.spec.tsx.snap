// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User renders 1`] = `
<DocumentFragment>
  <span
    class="user"
  >
    <span
      class="userName"
    >
      Antonio
    </span>
    <span
      class="jsx-134754294 avatar "
    >
      <span
        class="jsx-134754294 avatar-text"
      >
        A
      </span>
    </span>
  </span>
</DocumentFragment>
`;

exports[`User renders with image 1`] = `
<DocumentFragment>
  <span
    class="user"
  >
    <span
      class="userName"
    >
      Antonio
    </span>
    <span
      class="jsx-134754294 avatar "
    >
      <img
        class="jsx-134754294 avatar-img"
        src="https://react.geist-ui.dev/images/avatar.png"
      />
    </span>
  </span>
</DocumentFragment>
`;

exports[`User renders with image, reverse and large 1`] = `
<DocumentFragment>
  <span
    class="user"
  >
    <span
      class="userName userLarge"
    >
      Antonio
    </span>
    <span
      class="jsx-134754294 avatar "
    >
      <img
        class="jsx-134754294 avatar-img"
        src="https://react.geist-ui.dev/images/avatar.png"
      />
    </span>
  </span>
</DocumentFragment>
`;

exports[`User renders with name reverse 1`] = `
<DocumentFragment>
  <span
    class="user userReverse"
  >
    <span
      class="userName"
    >
      Antonio
    </span>
    <span
      class="jsx-134754294 avatar "
    >
      <span
        class="jsx-134754294 avatar-text"
      >
        A
      </span>
    </span>
  </span>
</DocumentFragment>
`;

exports[`User renders with size large 1`] = `
<DocumentFragment>
  <span
    class="user"
  >
    <span
      class="userName userLarge"
    >
      Antonio
    </span>
    <span
      class="jsx-134754294 avatar "
    >
      <span
        class="jsx-134754294 avatar-text"
      >
        A
      </span>
    </span>
  </span>
</DocumentFragment>
`;
